std-bits/table/src/main/java/org/enso/table/data/column/storage/MixedStorage.java

```diff
@@ -242,4 +255,24 @@ public Storage<?> tryGettingMoreSpecializedStorage() {
     var inferredStorage = getInferredStorage();
     return inferredStorage != null ? inferredStorage : this;
   }
+
+  /**
+   * Checks if any numbers are large enough for the column to require formatin in the table viz.
+   *
+   * @return true/false if formatting is required
+   */
+  public Boolean cachedNumericFormatCheck() throws InterruptedException {
+    if (isNumericFormatRequired.isCancelled()) {
+      // Need to recompute the value, as was cancelled.
+      isNumericFormatRequired =
+          CompletableFuture.supplyAsync(() -> RequiresNumberFormatting.compute(this, null));
```

Comment:
At this point IMO there is no point in computing this on a background thread (`supplyAsync`). The next line of code is `isNumericFormatRequired.get()` which will block the current thread until that computation finishes.

Instead of blocking 2 threads for this single computation, IMO it is better to just perform the computation on the main thread at this point.

So

```suggestion
          CompletableFuture.completedFuture(RequiresNumberFormatting.compute(this, null));
```